sdk/src/asset_handlers/tiff_io.rs

```diff
@@ -0,0 +1,1546 @@
+
+impl CAILoader for TiffIO {
+    fn read_cai(&self, asset_reader: &mut dyn CAIRead) -> Result<Vec<u8>> {
+        let cai_data = get_cai_data(asset_reader)?;
+        Ok(cai_data)
+    }
+
+    fn read_xmp(&self, asset_reader: &mut dyn CAIRead) -> Option<String> {
+        let xmp_data = get_xmp_data(asset_reader)?;
+        String::from_utf8(xmp_data).ok()
+    }
+}
+
+impl AssetIO for TiffIO {
+    fn asset_patch_ref(&self) -> Option<&dyn AssetPatch> {
+        Some(self)
+    }
+    fn read_cai_store(&self, asset_path: &std::path::Path) -> Result<Vec<u8>> {
+        let mut reader = std::fs::File::open(asset_path)?;
+
+        self.read_cai(&mut reader)
+    }
+
+    fn save_cai_store(&self, asset_path: &std::path::Path, store_bytes: &[u8]) -> Result<()> {
+        let mut temp_file = Builder::new()
+            .prefix("c2pa_temp")
+            .rand_bytes(5)
+            .tempfile()?;
+
+        let mut reader = std::fs::File::open(asset_path)?;
+
+        let l = u64::value_from(store_bytes.len())
+            .map_err(|_err| Error::InvalidAsset("value out of range".to_string()))?;
+
+        let entry = IfdClonedEntry {
+            entry_tag: C2PA_TAG,
+            entry_type: C2PA_FIELD_TYPE,
+            value_count: l,
+            value_bytes: store_bytes.to_vec(),
+        };
+
+        tiff_clone_with_tags(&mut temp_file, &mut reader, vec![entry])?;
+
+        std::fs::copy(temp_file.path(), asset_path)?;
+
+        Ok(())
+    }
+
+    fn get_object_locations(
+        &self,
+        asset_path: &std::path::Path,
+    ) -> Result<Vec<crate::asset_io::HashObjectPositions>> {
+        add_required_tags(asset_path)?;
+
+        let mut asset_reader =
+            std::fs::File::open(asset_path).map_err(|_err| Error::EmbeddingError)?;
+
+        let (idfs, first_idf_token, e, big_tiff) = map_tiff(&mut asset_reader)?;
+
+        let cai_ifd_entry = match idfs[first_idf_token].data.get_tag(C2PA_TAG) {
+            Some(ifd) => ifd,
+            None => return Ok(Vec::new()),
+        };
+
+        // make sure data type is for unstructured data
+        if cai_ifd_entry.entry_type != C2PA_FIELD_TYPE {
+            return Err(Error::InvalidAsset(
+                "Ifd entry for C2PA must be type UNKNOWN(7)".to_string(),
+            ));
+        }
+
+        let decoded_offset = decode_offset(cai_ifd_entry.value_offset, e, big_tiff)?;
+        let manifest_offset = usize::value_from(decoded_offset)
+            .map_err(|_err| Error::InvalidAsset("TIFF/DNG out of range".to_string()))?;
+        let manifest_len = usize::value_from(cai_ifd_entry.value_count)
+            .map_err(|_err| Error::InvalidAsset("TIFF/DNG out of range".to_string()))?;
+
+        Ok(vec![HashObjectPositions {
+            offset: manifest_offset,
+            length: manifest_len,
+            htype: HashBlockObjectType::Cai,
+        }])
+    }
+
+    fn remove_cai_store(&self, _asset_path: &std::path::Path) -> Result<()> {
+        Ok(())
```

Comment:
Really seems like it should return an error if it isn't supported

Reply:
I will add support.  I just overlooked this case.